language-design.md

```diff
@@ -17,7 +17,17 @@ For each quote, describe what it was about the quote that led you pick it.
 
 **Response**
 
+"...Perl, a major programming language used by untold zillions of developers, is no more intuitive to novices than a language with a randomly generated syntax." [Pavlus, 2012]
 
+I definitely disagreed with Pavlus' article, especially the claim quoted above. From my experience on the OCM project, I know that it can be hard to do studies on languages. Even though OCM was better from the standpoint of someone who was actually acquainted with parallel programming, it was hard to use novices to show this. User studies over the years have been problematic and it's very hard to teach people parallel programming over the course of half an hour or so. There are also a lot of problems with the idea of using novices to judge a language when they make up a very small portion of the userbase. Not all languages are designed to be easy to learn or to be very user friendly. In fact, I think it would be counterintuitive for most programming languages to have ease of use for novices as a primary concern. User-friendly languages will have to be more verbose than regular languages and will make doing just basic tasks a chore. In the example given in the article, a for loop in Quorum will take twice as many lines as C and about double to triple the number of characters. While languages like Quorum or Scratch may be more accessible to novices, they would likely hinder the productivity of most programmers in the long run. It's also worth noting that the paper only used a sample size of 18 people whose average age was 21 [[Stefik et al., 2011, p 5](http://ecs.victoria.ac.nz/foswiki/pub/Events/PLATEAU/2011Program/plateau2011-stefik.pdf)]. Even if the results were statistically significant, 18 people is a pretty small sample size. Their sample also probably don't represent many of the people learning to code, especially since many coding novices are significantly younger than 21.
```

Comment:
Indeed, as the Stefik _et al._ paper mentions, the small sample size is a threat to the validity of the study. That doesn't mean that the research methodology is incorrect or fundamentally flawed in some way. Lots of fields use qualitative / case study methods to investigate a question. In PL design, this is actually a really solid study that asks more questions than it answers and points the way to future research.